src/components/structures/UserSettings.js

```diff
@@ -152,10 +156,85 @@ module.exports = React.createClass({
         this.logoutModal.closeDialog();
     },
 
+    onEnableNotificationsChange: function(event) {
+        UserSettingsStore.setEnableNotifications(event.target.checked);
+    },
+
+    onAddThreepidClicked: function(value, shouldSubmit) {
+        if (!shouldSubmit) return;
+        var ErrorDialog = sdk.getComponent("dialogs.ErrorDialog");
+        var QuestionDialog = sdk.getComponent("dialogs.QuestionDialog");
+
+        var email_address = this.refs.add_threepid_input.value;
+        if (!Email.looksValid(email_address)) {
+            Modal.createDialog(ErrorDialog, {
+                title: "Invalid Email Address",
+                description: "This doesn't appear to be a valid email address",
+            });
+            return;
+        }
+        this.add_threepid = new AddThreepid();
+        // we always bind emails when registering, so let's do the
+        // same here.
+        this.add_threepid.addEmailAddress(email_address, true).then(() => {
```

Comment:
We want to terminate the promise chain here so `s/then/done/` so we throw uncaught errors.